# EDA con Enfoque en CAC y LTV

Este cuaderno se centra en **preguntas de negocio** que generan valor al entender la relación entre:
- **CAC**: Costo de Adquisición de Cliente.
- **LTV**: Valor de Vida del Cliente.

Adicionalmente, explora aspectos clave como **suscripción**, **descuentos**, **categoría** del producto, **frecuencia de compras**, **método de pago**, etc. Con estas preguntas podrás tomar **decisiones estratégicas** sobre tu modelo de negocio, campañas de marketing y retención de clientes.

## Contenido:
1. **Importar librerías y cargar datos**
2. **Inspección inicial** (dimensiones, tipos de datos, nulos)
3. **Limpieza básica** (si es necesaria)
4. **Creación de columna** ratio `LTV/CAC` (rentabilidad por cliente)
5. **Preguntas de negocio** (groupby, filtros, etc.)
6. **Conclusiones**

In [3]:
!wget !wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv

--2025-03-06 00:41:47--  http://!wget/
Resolving !wget (!wget)... failed: Name or service not known.
wget: unable to resolve host address ‘!wget’
--2025-03-06 00:41:47--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv
Resolving github.com (github.com)... 140.82.112.4
Connecting to github.com (github.com)|140.82.112.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv [following]
--2025-03-06 00:41:47--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HT

## 1. Importar librerías y cargar datos
Asegúrate de tener tu archivo CSV con las columnas mencionadas: `CAC`, `LTV`, `Purchase Amount (USD)`, etc. Ajusta la ruta según tu archivo real.

In [4]:
import pandas as pd
import numpy as np


# Cargar el dataset
df = pd.read_csv("shopping_trends_LTV_CAC.csv")


# Mostramos las primeras filas para verificar
df.head()

Customer ID  Age Gender Item Purchased  Category  Purchase Amount (USD)  \
0            1   55   Male         Blouse  Clothing                     53   
1            2   19   Male        Sweater  Clothing                     64   
2            3   50   Male          Jeans  Clothing                     73   
3            4   21   Male        Sandals  Footwear                     90   
4            5   45   Male         Blouse  Clothing                     49   

        Location Size      Color  Season  ...  Shipping Type Discount Applied  \
0       Kentucky    L       Gray  Winter  ...        Express              Yes   
1          Maine    L     Maroon  Winter  ...        Express              Yes   
2  Massachusetts    S     Maroon  Spring  ...  Free Shipping              Yes   
3   Rhode Island    M     Maroon  Spring  ...   Next Day Air              Yes   
4         Oregon    M  Turquoise  Spring  ...  Free Shipping              Yes   

  Promo Code Used Previous Purchases Preferred Payment Method  \
0             Yes                 14                    Venmo   
1             Yes                  2                     Cash   
2             Yes                 23              Credit Card   
3             Yes                 49                   PayPal   
4             Yes                 31                   PayPal   

  Frequency of Purchases    CAC   LTV Rentabilidad  LTV_CAC_Ratio  
0            Fortnightly  112.0  1512    16.434783      13.500000  
1            Fortnightly   95.0  1497    19.960000      15.757895  
2                 Weekly   32.0  1137    94.750000      35.531250  
3                 Weekly    NaN  1914    26.957746      26.957746  
4               Annually   47.0  1300    48.148148      27.659574  

[5 rows x 23 columns]

## 2. Inspección inicial
1. **Dimensiones (filas x columnas)**
2. **Info general**: tipos de dato, nulos, etc.
3. **Valores nulos** por columna.

Si existen valores nulos en columnas importantes (por ejemplo, `CAC`, `LTV`, `Customer ID`), evalúa si los eliminarás o imputarás.

In [5]:

print("Dimensiones del DataFrame:", df.shape)

print("\nInformación general del DataFrame:")
print(df.info())


print("\nValores nulos por columna:")
print(df.isnull().sum())


Dimensiones del DataFrame: (3900, 23)

Información general del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3900 entries, 0 to 3899
Data columns (total 23 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Customer ID               3900 non-null   int64  
 1   Age                       3900 non-null   int64  
 2   Gender                    3900 non-null   object 
 3   Item Purchased            3900 non-null   object 
 4   Category                  3900 non-null   object 
 5   Purchase Amount (USD)     3900 non-null   int64  
 6   Location                  3900 non-null   object 
 7   Size                      3900 non-null   object 
 8   Color                     3900 non-null   object 
 9   Season                    3900 non-null   object 
 10  Review Rating             3900 non-null   float64
 11  Subscription Status       3900 non-null   object 
 12  Payment Method            3900 non-null   ob

## 3. Limpieza básica
Por ejemplo, si detectas `CAC <= 0` o `LTV <= 0` y consideras que no tienen sentido en tu negocio, podrías filtrar esos registros. Ajusta según tu criterio.

In [6]:
df[(df['CAC']<=0) | (df['LTV']<=10)]

Empty DataFrame
Columns: [Customer ID, Age, Gender, Item Purchased, Category, Purchase Amount (USD), Location, Size, Color, Season, Review Rating, Subscription Status, Payment Method, Shipping Type, Discount Applied, Promo Code Used, Previous Purchases, Preferred Payment Method, Frequency of Purchases, CAC, LTV, Rentabilidad, LTV_CAC_Ratio]
Index: []

[0 rows x 23 columns]

## 4. Creación de columna: Ratio LTV/CAC
Es una métrica clave de la **rentabilidad** del cliente. Un valor > 1 indica que, en teoría, el cliente retorna más de lo que cuesta adquirirlo.

In [7]:
# Crear una nueva columna rentabilidad cliente
df['LTV/CAC'] = df['LTV'] / df['CAC']


## 5. Preguntas de Negocio
En vez de usar gráficas o correlaciones, nos centramos en **preguntas que generen valor** al análisis. Cada una se resuelve con **groupby, filtros o estadísticas** simples.

1. **¿Cuántos clientes únicos hay y cuál es la edad promedio?**
2. **¿Cuál es el CAC promedio y el LTV promedio en todo el dataset?**
3. **¿Cuál es la ratio LTV/CAC promedio?**
4. **¿Qué categoría tiene el mayor LTV en promedio?**
5. **¿En cuál categoría se justifica más el CAC?** (categoría con mayor ratio LTV/CAC)
6. **¿Los clientes suscritos tienen LTV mayor que los no suscritos?**
7. **¿Cuál es el método de pago más frecuente y su LTV promedio?**
8. **¿Quiénes usan descuentos con mayor frecuencia?** (Por ejemplo, ¿cierto género, cierto rango de edad?)
9. **¿Cuántos clientes aprovechan códigos promocionales (Promo Code Used)?** y su ratio LTV/CAC
10. **¿Cuál es la frecuencia de compra más común (Frequency of Purchases)?** y su LTV asociado.
11. **¿Cuál es la ubicación (Location) con mayor ratio LTV/CAC?**
12. **Entre los clientes con `Previous Purchases` > X, cuál es el gasto promedio y LTV promedio?**


### 5.1 ¿Cuántos clientes únicos hay y cuál es la edad promedio?

In [8]:
df['Age'].mean()


44.06846153846154

In [9]:
df['Customer ID'].nunique()

3900

### 5.2 ¿Cuál es el CAC promedio y el LTV promedio en todo el dataset?

In [10]:

print("CAC promedio:", df['CAC'].mean())
print("LTV promedio:", df['LTV'].mean())


CAC promedio: 66.29233565586186
LTV promedio: 1108.9448717948717


### 5.3 ¿Cuál es la ratio LTV/CAC promedio?

In [11]:
print("CAC ratio promedio:", df['CAC'].mean())
print("LTV ratio promedio:", df['LTV'].mean())


CAC ratio promedio: 66.29233565586186
LTV ratio promedio: 1108.9448717948717


### 5.4 ¿Qué categoría tiene el mayor LTV en promedio?

In [12]:

print(df.groupby('Category')['LTV'].mean().idxmax())


Accessories


### 5.5 ¿En cuál categoría se justifica más el CAC? (mayor ratio LTV/CAC)

In [14]:

category_ltv_cac = df.groupby('Category')['LTV/CAC'].mean()


print("La categoría con mayor ratio LTV/CAC es: {best_category}")


La categoría con mayor ratio LTV/CAC es: {best_category}


### 5.6 ¿Los clientes suscritos tienen LTV mayor que los no suscritos?

In [15]:
df.groupby('Subscription Status')['LTV'].mean()

Subscription Status
No     1113.425009
Yes    1096.831909
Name: LTV, dtype: float64

no los clientes suscritos tienen 1096.831909 en cambio los no suscritos tienen 1113.425009


### 5.7 ¿Cuál es el método de pago más frecuente y su LTV promedio?

In [16]:
df.groupby('Payment Method')['LTV'].mean()

Payment Method
Bank Transfer    1139.090190
Cash             1151.208333
Credit Card      1097.510057
Debit Card       1089.439179
PayPal           1087.108150
Venmo            1090.260337
Name: LTV, dtype: float64

cash 1151.208333

### 5.8 ¿Quiénes usan descuentos con mayor frecuencia?
Podemos ver si hay algún grupo (por género, rango de edad, etc.) que destaca. A modo de ejemplo, segmentaremos por género:

In [17]:

discount_frequency_by_gender = df.groupby('Gender')['Discount Applied'].sum()

gender_with_highest_discount_frequency = discount_frequency_by_gender.idxmax()

print(f"El género que usa descuentos con mayor frecuencia es: {gender_with_highest_discount_frequency}")


El género que usa descuentos con mayor frecuencia es: Male


Male	Yes	1677



```
# Esto tiene formato de código
```



Podrías repetir el ejercicio segmentando por **Age** (definiendo rangos) o por **Location**.

### 5.9 ¿Cuántos clientes aprovechan códigos promocionales y cuál es su ratio LTV/CAC?
El uso de promos puede modificar la rentabilidad.

In [ ]:
df

In [ ]:
df.groupby('Promo Code Used')['LTV/CAC'].mean()

Promo Code Used
No     22.252655
Yes    22.705036
Name: LTV/CAC, dtype: float64

clientes aprovechan códigos promocionales 22.705036

### 5.10 ¿Cuál es la frecuencia de compra más común (Frequency of Purchases) y su LTV?
Esto ayuda a ver si, por ejemplo, los clientes que compran semanalmente tienen un LTV más alto.

In [ ]:
df.groupby('Frequency of Purchases')['LTV'].mean().sort_values(ascending=False)

Frequency of Purchases
Bi-Weekly         1182.279707
Weekly            1111.230056
Annually          1107.472028
Fortnightly       1106.501845
Quarterly         1098.229130
Monthly           1087.001808
Every 3 Months    1072.965753
Name: LTV, dtype: float64

Bi-Weekly	1182.279707

### 5.11 ¿Cuál es la ubicación (Location) con mayor ratio LTV/CAC?

In [18]:
df.groupby('Location')['LTV/CAC'].mean().sort_values(ascending=False).head(1)


Location
New York    32.785849
Name: LTV/CAC, dtype: float64

New York	32.785849

### 5.12 Entre los clientes con `Previous Purchases` > X, ¿cuál es el gasto promedio y el LTV promedio?
Define un umbral X según tu contexto. Por ejemplo, X = 20:

In [ ]:
df[df['Previous Purchases'] > 20].agg({'Purchase Amount (USD)': 'mean', 'LTV': 'mean'})

Purchase Amount (USD)      59.697734
LTV                      1114.299273
dtype: float64

## 6. Conclusiones

En este Notebook, hemos abordado un **EDA orientado a negocio** en el que:
1. **Identificamos** la base total de clientes y su edad promedio.
2. **Calculamos** el CAC y LTV promedio, y la ratio LTV/CAC global.
3. **Segregamos** por categorías, suscripción, uso de descuentos, métodos de pago, etc.
4. **Resolvimos** preguntas de negocio que ayudan a:
   - **Identificar** cuáles categorías o ubicaciones ofrecen mejor rentabilidad (ratio LTV/CAC).
   - **Entender** si la suscripción eleva el LTV.
   - **Explorar** la efectividad de promos y descuentos.
   - **Diferenciar** frecuencias de compra y ver su impacto en LTV.
   - **Focalizar** inversiones de marketing en los segmentos más prometedores.

Este tipo de **análisis** proporciona las bases para **estrategias** de retención, adquisición y segmentación de clientes. ¡Espero que te sea útil para **generar valor** en tu negocio!